### ISSUE TITLE
ErrorException: Array and string offset access syntax with curly braces is deprecated

### ISSUE THREAD

**Issue by letumfalx** [gugglegum/memory-size#1]
Good day! 
Just reporting that I'm getting the following error when using this package.

```json
{
    "class": "ErrorException",
    "message": "Array and string offset access syntax with curly braces is deprecated",
    "code": 0,
    "file": "/home/.../app/vendor/gugglegum/memory-size/src/Formatter.php:106",
    "trace": [
        "..."
    ]
}
```

It is just a PHP notice for `PHP v7.3` and is now an error on `PHP v7.4+` .

Upon inspecting the file of interest, it looks like you just only need to change the way to access the array from curly brackets to square brackets without breaking anything.

Thanks and hoping that you can fix this up!

**Comment by gugglegum** [comment]
@letumfalx thanks for this report. I will fix it shortly. Sorry for inconvenience.

**Comment by gugglegum** [comment]
@letumfalx I fixed it in 1.0.2, could you confirm please?

**Comment by letumfalx** [comment]
@gugglegum thanks for the fix, tried `v1.0.2` on `PHP v7.4`, and there is no more error on my end.

By the way, I attempted to test it on a project using `PHP v8.0` but I got some error when `composer require`-ing it. It looks like the package cannot be used yet for `PHP v8.0` based on the `composer.json` file. Do I need to create a new issue to request a support for `PHP v8.0`? Thanks.

**Comment by gugglegum** [comment]
@letumfalx As you wish. I will check it out anyway. I just didn't tried PHP 8.0 yet.

**Comment by gugglegum** [comment]
@letumfalx I fixed Composer issue when using PHP 8.0 in `v1.0.3`.

**Comment by letumfalx** [comment]
@gugglegum tried on my end, and everything works properly. Thanks for your time fixing the issues. I will be closing the issue now.
